src/test/java/de/berlin/lostberlin/service/BusinessServiceTest.java

```diff
@@ -9,106 +8,55 @@
 import de.berlin.lostberlin.model.business.persistence.Business;
 import de.berlin.lostberlin.repository.BusinessRepository;
 import org.junit.Before;
-import org.junit.Ignore;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.springframework.boot.test.context.ConfigFileApplicationContextInitializer;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.test.context.SpringBootTest;
-import org.springframework.boot.test.mock.mockito.MockBean;
-import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit4.SpringRunner;
 
-import javax.validation.ValidatorFactory;
 import java.util.List;
-import java.util.Optional;
 
-import static java.util.Collections.singletonList;
-import static org.assertj.core.api.Assumptions.assumeThat;
 import static org.junit.Assert.*;
-import static org.mockito.BDDMockito.given;
-import static org.mockito.Mockito.times;
-import static org.mockito.Mockito.verify;
-
-/*
-Unfortunately, I get the same problem with @JpaAuditing annotation even after it's been externalized to ApplicationConfig file. So I had to comment it out for making tests.
-I added @ContextConfiguration because ApplicationContext could not be loaded without it.
- */
 
 @RunWith(SpringRunner.class)
-@ContextConfiguration(classes = ApplicationConfig.class, initializers = ConfigFileApplicationContextInitializer.class)
-@SpringBootTest
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
 public class BusinessServiceTest {
```

Comment:
I still cannot make it work since it creates MailService beans as well (which we do not need for the test) and cannot find sendgrid.properties file.

Reply:
Damn, seems to work, bloody stuff. See pr.

Reply:
Just opened one